apps/entity-service/src/entities/processors/media.processor.ts

```diff
@@ -0,0 +1,245 @@
+  readonly DTO = MediaDto;
+
+  constructor(
+    protected readonly entityType: EntityType,
+    protected readonly entityUuid: string,
+    protected readonly entityModelClass: EntityClass<EntityModel>,
+    protected readonly entitiesService: EntitiesService
+  ) {
+    super(entityType, entityUuid, entityModelClass, entitiesService);
+  }
+
+  private async getSiteModels(site: Site): Promise<QueryModelType[]> {
+    const models: QueryModelType[] = [{ modelType: this.entityModelClass.LARAVEL_TYPE, subquery: [site.id] }];
+    const subquery = Subquery.select(SiteReport, "id").eq("siteId", site.id);
+    models.push({ modelType: SiteReport.LARAVEL_TYPE, subquery: subquery.literal });
+    return models;
+  }
+
+  private async getNurseryModels(nursery: Nursery) {
+    const models: QueryModelType[] = [{ modelType: this.entityModelClass.LARAVEL_TYPE, subquery: [nursery.id] }];
+    const subquery = Subquery.select(NurseryReport, "id").eq("nurseryId", nursery.id);
+    models.push({ modelType: NurseryReport.LARAVEL_TYPE, subquery: subquery.literal });
+    return models;
+  }
+
+  private async getProjectModels(project: Project) {
+    const models: QueryModelType[] = [{ modelType: this.entityModelClass.LARAVEL_TYPE, subquery: [project.id] }];
+
+    const projectReportSubquery = Subquery.select(ProjectReport, "id").eq("projectId", project.id);
+    models.push({ modelType: ProjectReport.LARAVEL_TYPE, subquery: projectReportSubquery.literal });
+
+    const siteSubquery = Subquery.select(Site, "id").eq("projectId", project.id);
+    models.push({ modelType: Site.LARAVEL_TYPE, subquery: siteSubquery.literal });
+
+    const sites = await Site.findAll({ where: { projectId: project.id }, attributes: ["id"] });
+
+    const siteReportSubquery = Subquery.select(SiteReport, "id").in(
+      "siteId",
+      sites.map(site => site.id)
+    );
+    models.push({ modelType: SiteReport.LARAVEL_TYPE, subquery: siteReportSubquery.literal });
+
+    const nurseries = await Nursery.findAll({ where: { projectId: project.id }, attributes: ["id"] });
+    const nurserySubquery = Subquery.select(Nursery, "id").eq("projectId", project.id);
+    models.push({ modelType: Nursery.LARAVEL_TYPE, subquery: nurserySubquery.literal });
+
+    const nurseryReportSubquery = Subquery.select(NurseryReport, "id").in(
+      "nurseryId",
+      nurseries.map(nursery => nursery.id)
+    );
+    models.push({ modelType: NurseryReport.LARAVEL_TYPE, subquery: nurseryReportSubquery.literal });
+
+    return models;
+  }
+
+  private async getProjectReportModels(projectReport: ProjectReport) {
+    const models: QueryModelType[] = [{ modelType: this.entityModelClass.LARAVEL_TYPE, subquery: [projectReport.id] }];
+
+    const projectReportWithDue = await ProjectReport.findOne({
+      where: { id: projectReport.id },
+      attributes: ["dueAt"]
+    });
+
+    const project = await Project.findOne({ where: { id: projectReport.projectId }, attributes: ["id"] });
```

Comment:
This is another red flag: I can't think of any reason why you'd need to go to the DB to only pull the ID field when you in fact already have the ID in memory. The two lines below can simply use `eq("projectId", projectReport.projectId)`

Reply:
you are right, removed